src/python/cli_new/lib/cli/mesos.py

```diff
@@ -504,13 +508,23 @@ def _attach_container_output(self):
         client from the agent.
         """
 
+        # Set authentication header
+        auth = None
+        # pylint: disable=line-too-long
+        if self.config.agent_principal() is not None and self.config.agent_secret() is not None:
+            auth = requests.auth.HTTPBasicAuth(
+                self.config.agent_principal(),
+                self.config.agent_secret()
+            )
+
```

Comment:
Given that this code occurs three times, it is definitely worth extracting into a dedicated function.

Reply:
U are right. I also moved it into the config.py. but I'm unsteady, maybe it makes more sense in http.py. :thinking: